code/modules/plumbing/plumbers/pill_press.dm

```diff
@@ -147,12 +163,22 @@
 				product_name = formatted_name
 		if("change_product")
 			product = params["product"]
+			var/max_volume
 			if (product == "pill")
-				max_volume = max_pill_volume
+				max_volume = MAX_PILL_VOLUME
 			else if (product == "patch")
-				max_volume = max_patch_volume
+				max_volume = MAX_PATCH_VOLUME
 			else if (product == "bottle")
-				max_volume = max_bottle_volume
-			current_volume = clamp(current_volume, min_volume, max_volume)
+				max_volume = MAX_BOTTLE_VOLUME
+			else
+				return
+			current_volume = clamp(current_volume, MIN_VOLUME, max_volume)
 		if("change_patch_style")
```

Comment:
Does UI really need all these actions? In chem master I just select the container by ref, without caring whether it's pill, patch or a bottle.

```
	if(action == "selectContainer")
		selected_container = params["ref"]
		return TRUE
```

And then you can get the volume as `initial(container.volume)`

Reply:
the constants `MAX_PILL_VOLUME` & `MAX_PATCH_VOLUME` are required for the UI(which is why i send them as static data) to know how to clamp user input. As for using ref yup I'll look into it

Reply:
beautiful